modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/SignInState.kt

```diff
@@ -17,14 +19,14 @@ sealed class SignInState {
         get() = this is SignedIn && this.subscriptionStatus is SubscriptionStatus.Free
 
     val isSignedInAsPlus: Boolean
-        get() = this is SignedIn && this.subscriptionStatus is SubscriptionStatus.Plus
+        get() = this is SignedIn && this.subscriptionStatus is SubscriptionStatus.Plus && this.subscriptionStatus.type == SubscriptionType.PLUS
+
+    val isSignedInAsPatron: Boolean
+        get() = BuildConfig.ADD_PATRON_ENABLED && this is SignedIn && this.subscriptionStatus is SubscriptionStatus.Plus && this.subscriptionStatus.type == SubscriptionType.PATRON
```

Comment:
I should either rename `SubscriptionStatus.Plus` to `SubscriptionStatus.PlusOrPatron` differentiated by type or include another data class  `SubscriptionStatus.Patron` with `SubscriptionType.Patron` and keep `SubscriptionStatus.Plus` only for `SubscriptionType.Plus`.

https://github.com/Automattic/pocket-casts-android/blob/0c4e15cc507b62020cf4df43b4f8a02a4646fa80/modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/SubscriptionStatus.kt#L23

Lmk if you have a preference or I'll defer this decision until corresponding endpoint is updated.

Reply:
I would tend to favor separate types for Plus and Patron, but I don't feel strongly and if that's fighting against what the endpoint is doing it may not be worth it. Basically, I'm fine with whatever feels better to you.